precompile/contracts/nativeminter/nativemintertest/simulated_test.go

```diff
@@ -0,0 +1,303 @@
+	"github.com/ava-labs/subnet-evm/precompile/contracts/nativeminter"
+	"github.com/ava-labs/subnet-evm/precompile/contracts/testutils"
+	"github.com/ava-labs/subnet-evm/utils"
+
+	sim "github.com/ava-labs/subnet-evm/ethclient/simulated"
+)
+
+var (
+	adminKey, _        = crypto.GenerateKey()
+	unprivilegedKey, _ = crypto.GenerateKey()
+
+	adminAddress        = crypto.PubkeyToAddress(adminKey.PublicKey)
+	unprivilegedAddress = crypto.PubkeyToAddress(unprivilegedKey.PublicKey)
+)
+
+func TestMain(m *testing.M) {
+	// Ensure libevm extras are registered for tests.
+	core.RegisterExtras()
+	customtypes.Register()
+	params.RegisterExtras()
+	m.Run()
+}
+
+func newBackendWithNativeMinter(t *testing.T) *sim.Backend {
+	t.Helper()
+	chainCfg := params.Copy(params.TestChainConfig)
+	// Match the simulated backend chain ID used for signing (1337).
+	chainCfg.ChainID = big.NewInt(1337)
+	// Enable ContractNativeMinter at genesis with admin set to adminAddress.
+	params.GetExtra(&chainCfg).GenesisPrecompiles = extras.Precompiles{
+		nativeminter.ConfigKey: nativeminter.NewConfig(utils.NewUint64(0), []common.Address{adminAddress}, nil, nil, nil),
+	}
+	return sim.NewBackend(
+		types.GenesisAlloc{
+			adminAddress:        {Balance: big.NewInt(1000000000000000000)},
+			unprivilegedAddress: {Balance: big.NewInt(1000000000000000000)},
+		},
+		sim.WithChainConfig(&chainCfg),
+	)
+}
+
+// Helper functions to reduce test boilerplate
+
+func deployERC20NativeMinter(t *testing.T, b *sim.Backend, auth *bind.TransactOpts, initSupply *big.Int) (common.Address, *ERC20NativeMinter) {
+	t.Helper()
+	addr, tx, contract, err := DeployERC20NativeMinter(auth, b.Client(), nativeminter.ContractAddress, initSupply)
+	require.NoError(t, err)
+	testutils.WaitReceiptSuccessful(t, b, tx)
+	return addr, contract
+}
+
+func deployMinter(t *testing.T, b *sim.Backend, auth *bind.TransactOpts, tokenAddress common.Address) (common.Address, *Minter) {
+	t.Helper()
+	addr, tx, contract, err := DeployMinter(auth, b.Client(), tokenAddress)
+	require.NoError(t, err)
+	testutils.WaitReceiptSuccessful(t, b, tx)
+	return addr, contract
+}
+
+func verifyRole(t *testing.T, nativeMinter *INativeMinter, address common.Address, expectedRole allowlist.Role) {
+	t.Helper()
+	role, err := nativeMinter.ReadAllowList(nil, address)
+	require.NoError(t, err)
+	require.Equal(t, expectedRole.Big(), role)
+}
+
+func setAsEnabled(t *testing.T, b *sim.Backend, nativeMinter *INativeMinter, auth *bind.TransactOpts, address common.Address) {
+	t.Helper()
+	tx, err := nativeMinter.SetEnabled(auth, address)
+	require.NoError(t, err)
+	testutils.WaitReceiptSuccessful(t, b, tx)
+}
+
+func TestNativeMinter(t *testing.T) {
+	chainID := big.NewInt(1337)
+	admin := testutils.NewAuth(t, adminKey, chainID)
+
+	initSupply := big.NewInt(1000)
+	amount := big.NewInt(100)
+
+	type testCase struct {
+		name string
+		test func(t *testing.T, backend *sim.Backend, nativeMinterIntf *INativeMinter)
+	}
+
+	testCases := []testCase{
```

Comment:
all of these tests are 1:1 copies of the tests in https://github.com/ava-labs/subnet-evm/blob/c65752b631b07206d506111cd52e5d686a9f02b0/contracts/contracts/test/ERC20NativeMinterTest.sol

Reply:
Is the coverage still good now? post removal?